### PR TITLE
feat(kafka): Add json encoding to source

### PR DESCRIPTION
**Pull request by kuiperda**
### Proposed Change
On Kafka (OTLP) source:
- Add enum to choose encoding for log payload with options `otlp_proto`, `raw`, `text`, `json` - previously defaulted to `otlp_proto` and is updated to fulfill customer request to parse a json payload. 
- Add enum to choose encoding for trace payload
- Metrics unchanged (there is only the `otlp_proto` option which is the default)

- Receiver documentation [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/v0.81.0/receiver/kafkareceiver)

##### Checklist
- [x] Changes are tested
- [x] CI has passed

### CODE REVIEW

File: resources/source-types/kafka_otlp.yaml [hunk 1]
```diff
@@ -331,6 +331,21 @@ spec:
           operator: equals
           value: true
 
+    - name: log_encoding
+      label: Log Encoding
+      description: The encoding of the log payload received from Kafka.
```
Comment (jsirianni) [inline resources/source-types/kafka_otlp.yaml:336]: This is nit picky, but I think we should refer to kafka payloads as "Events". The receiver is a subscriber to the topic. 
```
To publish (write) and subscribe to (read) streams of events
```
https://kafka.apache.org/intro

```suggestion
      description: The encoding of the log event pulled from the Kafka topic.
```

File: resources/source-types/kafka_otlp.yaml [hunk 2]
```diff
@@ -346,6 +361,23 @@ spec:
           operator: equals
           value: true
 
+    - name: trace_encoding
+      label: Trace Encoding
+      description: The encoding of the trace payload received from Kafka.
```
Comment (jsirianni) [inline resources/source-types/kafka_otlp.yaml:366]: See comment from log_encoding

**Review by jsirianni — COMMENTED** [review]
Left two comments related to param descriptions.

I am not sure the best way to test this. You might be able to get with @antonblock and see how he has been pushing events to kafka topics.

**Review by jsirianni — APPROVED** [review]
👍

**Review by kuiperda — COMMENTED** [review]
Updated source per ticket review comments. 
- Source displayName and description updated to reflect more encoding options
- Is there any issue with renaming the <source>.yaml file and the metadata.name?

**Review by cpheps — APPROVED** [review]

**Pull request merged**